### PR TITLE
composition: allow arguments in `@requires` selection sets

### PR DESCRIPTION
**Pull request by tomhoule**
closes GB-6295

### CODE REVIEW

File: engine/crates/composition/src/subgraphs/keys.rs [hunk 1]
```diff
@@ -121,15 +139,16 @@ impl NestedKeyFields {
...
 pub(crate) struct Key {
     selection_set: Vec<Selection>,
     resolvable: bool,
 }
 
-#[derive(PartialEq, Eq, PartialOrd, Ord, Debug)]
+#[derive(PartialEq, PartialOrd, Debug)]
 pub(crate) struct Selection {
     pub(crate) field: StringId,
+    pub(crate) arguments: Vec<(StringId, Value)>,
```
Comment (Finistere) [inline engine/crates/composition/src/subgraphs/keys.rs:151]: Hum, unless I'm mistaken you're not validating the arguments, no?
IMHO composition should validate those and not bother with GraphQL coercion rules, it should just match.
Doesn't sound obvious to me that you'll be able to store it in a better format (`Vec<(InputValueDefinitionId, Value)>` typically) considering you're reading from SDL. But well

Though saying this but when providing a SDL schema in the CLI it'll bypass composition, so I'll have to validate in all cases. Errors at composition would be a lot friendlier for the user at least

**Review by Finistere — APPROVED** [review]
Nice

File: engine/crates/composition/src/subgraphs/keys.rs (hunk 1 above)
Comment (tomhoule) [reply engine/crates/composition/src/subgraphs/keys.rs:151]: Yes I agree it's probably easier to validate at composition time, and worth the extra effort to put an InputValueDefinitionId there. I'll work on that.

Comment (tomhoule) [reply engine/crates/composition/src/subgraphs/keys.rs:151]: Done & pushed, let's see what the tests say.

**Pull request merged**